phpBB/posting.php

```diff
@@ -44,6 +41,75 @@
 $submit = $request->is_set_post('post') && !$refresh && !$preview;
 $mode		= $request->variable('mode', '');
 
+// We need to choose which of the url parameters we must trust and ignore the others
+	$forum_id = 0;
+	$topic_id = 0;
+	$post_id = 0;
+
+switch ($mode)
+{
+	case 'popup':
+	case 'smilies':
+		$forum_id = $request->variable('f', 0);
+		break;
+
+	case 'post':
+		$forum_id = $request->variable('f', 0);
+		if (!$forum_id)
+		{
+			$user->setup('posting');
+			trigger_error('NO_FORUM');
+		}
+		break;
+
+	case 'bump':
+	case 'reply':
+		$topic_id = $request->variable('t', 0);
+		if ($topic_id)
+		{
+			$sql = 'SELECT forum_id
+				FROM ' . TOPICS_TABLE . "
+				WHERE topic_id = $topic_id";
+			$result = $db->sql_query($sql);
+			$forum_id = (int) $db->sql_fetchfield('forum_id');
+			$db->sql_freeresult($result);
+		}
+		if (!$topic_id || !$forum_id)
+		{
+			$user->setup('posting');
+			trigger_error('NO_TOPIC');
+		}
+		break;
+
+	case 'edit':
+	case 'delete':
+	case 'quote':
+	case 'soft_delete':
+		$post_id = $request->variable('p', 0);
+		if ($post_id)
+		{
+			$topic_forum = array();
+
+			$sql = 'SELECT t.topic_id, t.forum_id
+				FROM ' . TOPICS_TABLE . ' t, ' . POSTS_TABLE . ' p
+				WHERE p.post_id = ' . $post_id . '
+				AND t.topic_id = p.topic_id';
+			$result = $db->sql_query($sql);
+			$topic_forum = $db->sql_fetchrow();
+			$topic_id = (int) $topic_forum['topic_id'];
+			$forum_id = (int) $topic_forum['forum_id'];
+			$db->sql_freeresult($result);
+		}
+
+		if (!$post_id || !$topic_id || !$forum_id)
+		{
+			$user->setup('posting');
```

Comment:
Here the `$user->setup('posting')` actually is needed.

Reply:
Are you requesting a change here?

Reply:
No, this was just a note to highlight that this one should not be removed like the other occurences